File: EcoQuestAPI/src/main/java/com/proyecto/spring/dto/ComentarioDTO.java
package com.proyecto.spring.dto;

import com.proyecto.spring.modelos.Comentario;

public class ComentarioDTO {

    public Long id;
    public String texto;
    public String fechaHora;
    public Long usuarioId;
    public String usuarioNombre;
    public String usuarioNombreUsuario;
    public String usuarioImagen;
    public Long eventoId;
    public Long comentarioPadreId;
    public int cantidadRespuestas;

    public ComentarioDTO() {}

    public ComentarioDTO(Comentario c) {
        this.id = c.getId();
        this.texto = c.getTexto();
        this.fechaHora = c.getFechaHora() != null ? c.getFechaHora().toString() : null;
        this.usuarioId = c.getUsuario() != null ? c.getUsuario().getId() : null;
        this.usuarioNombre = c.getUsuario() != null ? c.getUsuario().getNombre() + " " + c.getUsuario().getApellido() : null;
        this.usuarioNombreUsuario = c.getUsuario() != null ? c.getUsuario().getNombreUsuario() : "Usuario";
        this.usuarioImagen = c.getUsuario() != null ? c.getUsuario().getImagen() : null;
        this.eventoId = c.getEvento() != null ? c.getEvento().getId() : null;
        this.comentarioPadreId = c.getComentarioPadreId();
    }
}

File: EcoQuestMobile/app/src/main/java/com/ecoquest/app/data/remote/dto/ComentarioDto.kt
package com.ecoquest.app.data.remote.dto

import com.google.gson.annotations.SerializedName

data class ComentarioDto(
    @SerializedName("id") val id: Long = 0,
    @SerializedName("texto") val texto: String = "",
    @SerializedName("fechaHora") val fechaHora: String = "",
    @SerializedName("usuarioId") val usuarioId: Long = 0,
    @SerializedName("usuarioNombre") val usuarioNombre: String = "",
    @SerializedName("usuarioNombreUsuario") val usuarioNombreUsuario: String = "",
    @SerializedName("usuarioImagen") val usuarioImagen: String? = null,
    @SerializedName("eventoId") val eventoId: Long = 0,
    @SerializedName("comentarioPadreId") val comentarioPadreId: Long? = null,
    @SerializedName("cantidadRespuestas") val cantidadRespuestas: Int = 0
)
